subxt/src/runtime_api/runtime_payload.rs

```diff
@@ -69,7 +69,11 @@ pub trait RuntimeApiPayload {
 #[derive(Derivative)]
 #[derivative(
     Clone(bound = "ArgsData: Clone"),
-    Debug(bound = "ArgsData: std::fmt::Debug")
+    Debug(bound = "ArgsData: std::fmt::Debug"),
+    Eq(bound = "ArgsData: std::cmp::Eq"),
+    Ord(bound = "ArgsData: std::cmp::Ord"),
+    PartialEq(bound = "ArgsData: std::cmp::PartialEq"),
```

Comment:
Does this generate:

```
impl<T: Eq> for Payload {}
impl<T: PartialEq> for Payload {}
impl<T: Ord> for Payload {}
```

Reply:
It does seem to generate that for the `Eq` (using cargo expand)
- ` impl<ArgsData, ReturnTy> ::std::cmp::Eq for Payload<ArgsData, ReturnTy> where ArgsData: std::cmp::Eq {}`. 

However, for the others it does match the fields 🤔

Reply:
> However, for the others it does match the fields 

What do you mean here? I'd be curious to see the generated output :)

Reply:
This is from the address of the constants:

```rust
impl<ReturnTy> ::std::cmp::PartialEq for Address<ReturnTy> {
            fn eq(&self, other: &Self) -> bool {
                true && match *self {
                    Address {
                        pallet_name: ref __self_0,
                        constant_name: ref __self_1,
                        constant_hash: ref __self_2,
                        _marker: ref __self_3,
                    } => match *other {
                        Address {
                            pallet_name: ref __other_0,
                            constant_name: ref __other_1,
                            constant_hash: ref __other_2,
                            _marker: ref __other_3,
                        } => {
                            true && &(*__self_0) == &(*__other_0)
                                && &(*__self_1) == &(*__other_1)
                                && &(*__self_2) == &(*__other_2)
                                && &(*__self_3) == &(*__other_3)
                        }
                    },
                }
            }
        }
```

It looks about right to me :D

Reply:
Oops, yep this is the `PartialEq` for this:

```rust
        impl<ArgsData, ReturnTy> ::std::cmp::PartialEq for Payload<ArgsData, ReturnTy>
        where
            ArgsData: std::cmp::PartialEq,
        {
            fn eq(&self, other: &Self) -> bool {
                true && match *self {
                    Payload {
                        trait_name: ref __self_0,
                        method_name: ref __self_1,
                        args_data: ref __self_2,
                        validation_hash: ref __self_3,
                        _marker: ref __self_4,
                    } => match *other {
                        Payload {
                            trait_name: ref __other_0,
                            method_name: ref __other_1,
                            args_data: ref __other_2,
                            validation_hash: ref __other_3,
                            _marker: ref __other_4,
                        } => {
                            true && &(*__self_0) == &(*__other_0)
                                && &(*__self_1) == &(*__other_1)
                                && &(*__self_2) == &(*__other_2)
                                && &(*__self_3) == &(*__other_3)
                                && &(*__self_4) == &(*__other_4)
                        }
                    },
                }
            }
```